packages/core/docs/integrate/integration-guide.md

```diff
@@ -103,58 +103,81 @@ Once you copied and renamed the boilerplate run `yarn dev` in your `theme` folde
+- `index.server.ts` - contains the creation of api client, for direct connection to the integraded platform
+- `index.ts` - main entrypoint, that contains everythins else, such as: types, helper functions etc.
+
 
 ```ts
-// api-client/src/configuration.js
-import getProduct from './api/getProduct';
-import getCategory from './api/getCategory';
-import { apiClientFactory } from '@vue-storefront/core';
+// index.client.ts
+import { apiProxyFactory } from '@vue-storefront/core';
 
-interface Config {
+const onCreate = (config) => {
   // ...
-}
+  return { config };
+};
 
-interface API {
-  // ..
-}
+const { createApiProxy, integrationPlugin } = apiProxyFactory({
+  tag: 'ct',
+  onCreate,
+  api: { isGuest }
+});
 
-interface OnSetup {
-  config: Config;
-  client: EcommerceAPI;
-}
+export {
+  createApiProxy,
+  integrationPlugin
+};
+```
+
+
+```ts
+// index.server.ts
+import * as api from './api';
+import { apiClientFactory } from '@vue-storefront/core';
 
-const onSetup = (config: Config): OnSetup => {
-  const client = new EcommerceAPI(config)
+const onCreate = (settings) => {
+  const config = { ..setings }
+  const client = new ApiConnection()
 
-  return {
-    config,
-    client
-  };
+  return { config, client };
 };
 
-const { createApiClient } = apiClientFactory<Config, API>({
+
+
+const { createApiClient, integrationPlugin } = apiClientFactory({
   tag: 'ct',
-  onSetup,
-  api: {
-    getProduct,
-    getCategory,
-  }
+  onCreate,
+  api,
+  extensions: []
 });
+
+export {
+  createApiClient,
+  integrationPlugin
+};
 ```
 
 ```ts
-// api-client/src/index.js
-import { createApiClient } from './configuration';
-
-export { createApiClient }
+// index.ts
+export * from './types/Api';
 ```
 
-To create `api-client` you need to call `apiClientFactory`. As parameters to this deliver a few things:
-- `tag` - that's the short name of your integration which will be used to distinguish it among of others
-- `onSetup` - an optional function that will be called during creating your API. In this place, you can call everything you need to create a connection to the API, such as creating SDK (eg. `axios` creation), merge given config with the defaults etc.
-- `api` - this is the section where you need to pass all of the API function you have created
+To create `api-client` instances you have to use correspoing factory, depending on what api you are creating: proxy or direct one. The creation in both cases in pretty similar, with small differences in the used fields:
+
+- `tag` - that's the short name of your integration which will be used to distinguish it among of others 
+- `onCreate` - a function that will be called during creating your API. In this place, you can call everything you need to create a connection to the API, such as creating SDK (eg. axios creation), merge given config with the defaults etc.. This function always returns `client` (connection you created) and `config` or (in case it's proxy) just `config`.
```

Comment:
eg -> `e.g.`

Reply:
docs will be updated in the separated pr